File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Control/Toggle.java
package org.firstinspires.ftc.teamcode.Control;

public class Toggle {
    boolean currState = false;
    boolean prevState = false;
    boolean taskState = true;

    public boolean toggle(boolean boolState){

        if(boolState){
            currState = true;
        }

        else{
            currState = false;
            if(prevState){
                taskState = !taskState;
            }
        }

        prevState = currState;

        return taskState;
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Control/Toggle.kt
package org.firstinspires.ftc.teamcode.Control

class Toggle {
    private var currState = false
    private var prevState = false
    private var taskState = true

    fun toggle(boolState: Boolean): Boolean {
        if (boolState) {
            currState = true
        } else {
            currState = false
            if (prevState) {
                taskState = !taskState
            }
        }

        prevState = currState

        return taskState
    }
}
